[HW-1] feat: Create a Node.js application using Express.js framowork …

yevheniia-balda opened the pull request:
Create a Node.js application using the Express.js framework and integrate Swagger documentation. The application should also have a health-check endpoint and be hosted on GitHub. It should be run using Nodemon. 

# Requirements
1. Initialize a Node.js project:
- Set up a new Node.js project using npm or yarn.
- Install the necessary dependencies, including Express.js and Nodemon.

2. Create an Express.js Application:
- Set up an Express.js application with a basic server.
- Create a route for a health-check endpoint (GET /health-check) that returns a JSON response indicating the server is running.

3.Integrate Swagger Documentation:
- Install and set up Swagger UI and Swagger JS Docs for generating API documentation.
- Document at least one API endpoint (GET /health-check) 

4. Host the Project on GitHub:
- Create a GitHub repository for your Node.js project.
- Push your project's source code and documentation to the GitHub repository.

5.Test Your Application:
- Use Nodemon to run your application locally.
- Use a tool like Postman or CURL to test your health-check endpoint and ensure it returns the expected response.
- Access the Swagger documentation to verify that your API documentation is generated correctly.

6. Additional Features (Optional, but encouraged):
- Add more endpoints to your Express.js application and document them using Swagger.
- Implement error handling for common HTTP status codes (e.g., 404 Not Found, 500 Internal Server Error) and document error responses in Swagger.

Commit:
[HW-1] feat: Create a Node.js application using Express.js framowork and Swagger documentation

Commit:
[HW-1] refactor: Update README.md file

Commit:
[HW-1] refactor: Update package-lock.json file

KryvDanylo commented on src/routes/index.ts:
```diff
@@ -0,0 +1,67 @@
+    this.app.get('/health-check', (req, res) => {
+      res.status(200).json({message: 'Server is running', app: 'Health'})
+    });
+    this.app.use('/health-check', [notFoundHandler, errorHandler] ),
```
You can set an error handler at the end of the init method once and it should handle all of errors from routes.
http://expressjs.com/en/guide/error-handling.html#error-handling

yevheniia-balda replied:
Simplified the middlewares' setup for the route

KryvDanylo commented on src/routes/index.ts:
```diff
@@ -0,0 +1,67 @@
+    this.app.get('/health', (req, res) => {
+      res.status(200).json({message: 'This is Home page'});
+    });
+    this.app.use('/health', [notFoundHandler, errorHandler] );
```
remove the error handler, look at the above

yevheniia-balda replied:
Simplified the middlewares' setup for the route

KryvDanylo commented on src/routes/index.ts:
```diff
@@ -0,0 +1,67 @@
+import { errorHandler } from "../middlewares/errorHandler.middleware";
+import { notFoundHandler } from "../middlewares/errorHandler.middleware";
+
+class AppRouter {
```
A good way, but if you have a lot of routes method init would be enormous and not readable. So I propose creating a health route file, setting up a route by default Express method, and adding it to this class. Also, all of the global middleware, swagger, and error handlers can be added here. If you need help, pls ping me.

yevheniia-balda replied:
Great advice, I used to do it this way before, but I wasn't sure if it was appropriate in this case. I'll make the changes and ping you to take a look :)

yevheniia-balda replied:
Moved health routes to the api/health.route.ts file .  I made some changes to the code, please take a look when you have time. Thank you for the comments :)

KryvDanylo commented on src/app.ts:
```diff
@@ -0,0 +1,32 @@
+  console.log(`App running on port ${PORT}`);
+});
+
+app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swaggerSpec));
```
Move it to AppRouter. See my commentary on the route.

yevheniia-balda replied:
Moved Swagger configuration related code to the AppRouter

Commit:
[HW-1] refactor: Update AppRouter

- Moved health routes to the api/health.route.ts file
- Moved Swagger configuration related code to the AppRouter
- Simplified the middlewares' setup for the route

KryvDanylo commented on src/routes/index.ts:
```diff
@@ -0,0 +1,37 @@
+      swaggerUi.serve,
+      swaggerUi.setup(this.swaggerSpec)
+    );
+    this.app.use("/", healthRouter, [notFoundHandler, errorHandler]);
```
So, I mean that you can write "this.app.use(errorHandler);" at the end of the "init" method and it should handle errors from all routes.

yevheniia-balda replied:
Moved the usage of error handlers to the end of the init() method - this.app.use([notFoundHandler, errorHandler]).

Commit:
[HW-1] refactor: update middlewares' setup for the routes

KryvDanylo reviewed (approved):
Good
